### PR TITLE
Implement bind / unbind handling in broker client + fake broker + tests

### PR DESCRIPTION
**Pull request by vaikas-google**
Address #263

### CODE REVIEW

File: pkg/brokerapi/openservicebroker/open_service_broker_client_test.go [hunk 1]
```diff
@@ -202,11 +210,120 @@ func TestDeprovisionInstanceAcceptedFailureAsynchronous(t *testing.T) {
...
+	if fbs.RequestObject == nil {
+		t.Errorf("BindingRequest was not received correctly")
+	}
+	actual := reflect.TypeOf(fbs.RequestObject)
+	expected := reflect.TypeOf(&brokerapi.BindingRequest{})
+	if actual != expected {
+		t.Errorf("Got the wrong type for the request, expected %v got %v", expected, actual)
+	}
+	received := fbs.RequestObject.(*brokerapi.BindingRequest)
+	if reflect.DeepEqual(*received, sent) {
```
Comment (arschles) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: can you pass `*sent` here?

Comment (vaikas-google) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: Yah, done. Also actually fixed the test condition :)

**Comment by vaikas-google** [comment]
Thanks, PTAL @arschles

File: pkg/brokerapi/openservicebroker/open_service_broker_client_test.go [hunk 2]
```diff
@@ -202,11 +210,120 @@ func TestDeprovisionInstanceAcceptedFailureAsynchronous(t *testing.T) {
...
+func TestBindOk(t *testing.T) {
+	fbs, fakeBroker := setup()
+	defer fbs.Stop()
+
+	c := NewClient(testBrokerName, fakeBroker.Spec.URL, "", "")
+
+	fbs.SetResponseStatus(http.StatusOK)
+	sent := &brokerapi.BindingRequest{}
+	if _, err := c.CreateServiceBinding(testServiceInstanceID, testServiceBindingID, sent); err != nil {
+		t.Error(err.Error())
```
Comment (duglin) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: just wondering, is it worth continuing in cases like this? or should it be t.Fatal() ?

Comment (duglin) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: same for rest of file

File: pkg/brokerapi/openservicebroker/open_service_broker_client_test.go (hunk 1 above)
Comment (duglin) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: I think you forgot this comment

File: pkg/brokerapi/openservicebroker/open_service_broker_client_test.go [hunk 3]
```diff
@@ -202,11 +210,120 @@ func TestDeprovisionInstanceAcceptedFailureAsynchronous(t *testing.T) {
...
+	if err := c.DeleteServiceBinding(testServiceInstanceID, testServiceBindingID); err != nil {
+		t.Error(err.Error())
+	}
+
+	verifyBindingMethodAndPath(http.MethodDelete, testServiceInstanceID, testServiceBindingID, fbs.Request, t)
+
+	if fbs.Request.ContentLength != 0 {
+		t.Errorf("not expecting a request body, but got one, size %d", fbs.Request.ContentLength)
+	}
+	expectPath := fmt.Sprintf(bindingSuffixFormatString, testServiceInstanceID, testServiceBindingID)
```
Comment (duglin) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: doesn't verifyBindingMethodAndPath check this already?

**Comment by arschles** [comment]
@vaikas-google LGTM

File: pkg/brokerapi/openservicebroker/open_service_broker_client_test.go (hunk 2 above)
Comment (vaikas-google) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: Changed all Error to Fatal throughout the file for consistency.

File: pkg/brokerapi/openservicebroker/open_service_broker_client_test.go (hunk 3 above)
Comment (vaikas-google) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: Yes, I refactored it and forgot the original code behind here :) Removed

File: pkg/brokerapi/openservicebroker/open_service_broker_client_test.go (hunk 1 above)
Comment (vaikas-google) [inline pkg/brokerapi/openservicebroker/open_service_broker_client_test.go]: Hm, I guess I didn't push the changes...

**Comment by duglin** [comment]
LGTM

**Comment by duglin** [comment]
just waiting on CI

**Comment by pmorie** [comment]
The light is green; the trap is clean! 👻

**Pull request merged**